src/core/text/TextStyle.js

```diff
@@ -755,3 +763,20 @@ function areArraysEqual(array1, array2)
 
     return true;
 }
+
+/**
+ * Utility function to ensure that object properties are copied by value, and not by reference
+ *
+ * @param {Object} target Target object to copy property into
+ * @param {Object} source Source object for the proporty to copy
+ * @param {string} prop Property name to copy
+ */
+function deepCopyProperty(target, source, prop) {
+    if (Array.isArray(source[prop])) {
+        target[prop] = source[prop].slice();
+    } else if (source[prop] && typeof source[prop] === 'object') {
+        Object.assign(target[prop], source[prop]);
```

Comment:
I don't think we need this `Object.assign` because no style properties are plain objects.

Reply:
Right now we don't, but in the past there have been PR's that have suggested having one, so that's why I kept it in just in case.

Happy to remove if we're sure 4.x will never have style properties has objects, and if we're changing how it works in 5.x

Reply:
Positive v4 will never have object style properties.